curator-recipes/src/main/java/org/apache/curator/framework/recipes/nodes/PersistentEphemeralNode.java

```diff
@@ -338,10 +338,14 @@ public void setData(byte[] data) throws Exception
         this.data.set(Arrays.copyOf(data, data.length));
         if ( isActive() )
         {
-            client.setData().inBackground().forPath(getActualPath(), this.data.get());
+            client.setData().inBackground().forPath(getActualPath(), getData());
         }
     }
 
+    byte[] getData() {
```

Comment:
I'd mark it as private for now and it can be exposed later if needed. If it were to be for testing we mark these as @VisibleForTesting to be explicit as to why it's package protected.